### PR TITLE
port findBreakingChanges

### PR DESCRIPTION
**Pull request by roippi**
For https://github.com/webonyx/graphql-php/issues/64.  cc https://github.com/webonyx/graphql-php/issues/169

Porting this was really just mechanical translation - kudos on making it so easy!

WIP because I've only done 1/3 of the tests or so.  The rest of the code should be ready for review though, in case we want to get a head start on that.

### CODE REVIEW

**Comment by vladar** [comment]
Awesome! I'll review a bit later and will merge as soon as you finish with tests. Thanks!

**Comment by vladar** [comment]
So is it ready for review/merge?

**Comment by roippi** [comment]
Yes

File: src/Utils/FindBreakingChanges.php [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+    )
+    {
+        $oldTypeMap = $oldSchema->getTypeMap();
+        $newTypeMap = $newSchema->getTypeMap();
+
+        $breakingChanges = [];
+        foreach ($oldTypeMap as $typeName => $typeDefinition) {
+            if (!isset($newTypeMap[$typeName])) {
+                $breakingChanges[] =
+                    ['type' => self::BREAKING_CHANGE_TYPE_REMOVED, 'description' => "${typeName} was removed."];
```
Comment (vladar) [inline src/Utils/FindBreakingChanges.php]: Had no idea that `${typeName}` is a valid form of variable interpolation in PHP. Cool %)

**Comment by vladar** [comment]
Thanks a lot! Really useful stuff!

**Pull request merged**